parser/encoding/request.go

```diff
@@ -0,0 +1,138 @@
+package encoding
+
+import (
+	"github.com/cockroachdb/errors"
+
+	"encr.dev/parser"
+	v1 "encr.dev/proto/encore/parser/schema/v1"
+)
+
+// ParamLocation is the request/response home of the parameter
+type ParamLocation int16
+
+const (
+	Header = 1 // Parameter is placed in the http header
+	Query  = 2 // Parameter is placed in the query string
+	Body   = 3 // Parameter is placed in the body
```

Comment:
These should specify the `ParamLocation` type (`Header ParamLocation = 1` etc) to make them typed constants